game/dino_jump_main.py

```diff
@@ -0,0 +1,394 @@
+        self.curr_walk = (
+            self.UPDATE_WALK_FRAMES
+        )  # counts how many frames before the legs switch
+
+        # Calculates Time Constants (physically realistic jumping)
+        self.TIME_TO_PEAK = self.JUMP_TIME / 2
+        self.ACCEL = (2 * self.JUMP_HEIGHT) / (self.TIME_TO_PEAK**2)
+        self.INIT_V = self.TIME_TO_PEAK * self.ACCEL
+        self.JUMP_COEF = self.JUMP_HEIGHT / (
+            self.TIME_TO_PEAK**2 - self.JUMP_TIME * self.TIME_TO_PEAK
+        )
+
+    def update_jumping_location(self):
+        # Using timing to determine if we are mid jump and y position should be adjusted
+        if pygame.time.get_ticks() - self.jump_start_time <= self.JUMP_TIME:
+            jump_time = pygame.time.get_ticks() - self.jump_start_time
+            self.y = self.OG_DINO_Y - self.JUMP_COEF * (
+                jump_time**2 - self.JUMP_TIME * jump_time
+            )
+        else:
+            if self.is_jumping:
+                self.jump_start_time = pygame.time.get_ticks()
+            else:
+                self.y = self.OG_DINO_Y
+
+    def update_frame(self):
+
+        # Determine when the dino foot change happens
+        if self.curr_walk <= 0:
+            self.is_right_foot_down = not self.is_right_foot_down
+            self.curr_walk = self.UPDATE_WALK_FRAMES
+        self.curr_walk -= 1
+
+        # Update dino pixel array
+        if (
+            self.is_jumping
+            or pygame.time.get_ticks() - self.jump_start_time < self.JUMP_TIME
+        ):
+            self.ARRAY = self.JUMP_ARRAY
+        elif self.is_ducking and self.is_right_foot_down:
+            self.ARRAY = self.RDUCK_ARRAY
+        elif self.is_ducking:
+            self.ARRAY = self.LDUCK_ARRAY
+        elif self.is_right_foot_down:
+            self.ARRAY = self.RRUN_ARRAY
+        else:
+            self.ARRAY = self.LRUN_ARRAY
+
+        # If the game is over, show the dead dino frame
+        if game.game_over:
+            self.ARRAY = self.DONE_ARRAY  # JOSH a new sprite array for the "done" look
+
+
+#############################################################################
+# VOID SETUP
+#############################################################################
+
+# Initialises the Dino Game
+game = Game()
+
+# Start Pygame
+pygame.init()
+game.draw_all()
+
+
+#############################################################################
+# VOID LOOP
+#############################################################################
+
+# Wait until first space bar press until the game starts
+while not game.start_game:
+    game.check_start_game()
+
+# Simulates the whole loop running
+while not game.game_over:
+    game.check_user_input()
+    game.dino.update_jumping_location()
+    game.all_obstacles.generate_obstacle()
+    game.dino.update_frame()
+    game.all_obstacles.update_location()
+    game.draw_all()
+
+print("game over!")  # JOSH everything after here is the end game state!

```

Comment:
nit: This comment can probably be dropped

Reply:
Dropped!